exercises/react/src/example/kotlin/React.kt

```diff
@@ -0,0 +1,65 @@
+class Reactor<T>() {
+    abstract inner class Cell {
+        abstract val value: T
+        internal val dependents = mutableListOf<ComputeCell>()
+    }
+
+    interface Subscription {
+        fun cancel()
+    }
+
+    inner class InputCell(initialValue: T) : Cell() {
+        override var value: T = initialValue
+            set(newValue) {
+                field = newValue
+                dependents.forEach { it.propagate() }
+                dependents.forEach { it.fireCallbacks() }
+            }
+    }
+
+    inner class ComputeCell private constructor(val newValue: () -> T) : Cell() {
+        override var value: T = newValue()
+            private set
+
+        private var lastCallbackValue = value
+        private var callbacksIssued = 0
+        private val activeCallbacks = mutableMapOf<Int, (T) -> Any>()
+
+        constructor(vararg cells: Cell, f: (List<T>) -> T) : this({ f(cells.map { it.value }) }) {
```

Comment:
`List` seemed most appropriate since it allowed indexing into it, so the lambdas can say `it[0]`, `it[0] + it[1]`, etc. Using `Iterable` *could* work, but would be weird - we can do `it.first` and `it.fold { a, b -> a + b}` or something, I suppose.